File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.java
import java.util.*;

public class Main {
    static int[] arr;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int M = sc.nextInt();
        sc.nextLine();
        arr = new int[N + 1];
        for (int i = 1; i <= N; i++) {
            arr[i] = i;
        }
        for (int i = 0; i < M; i++) {
            StringTokenizer st = new StringTokenizer(sc.nextLine(), " ");
            int from = Integer.parseInt(st.nextToken());
            int to = Integer.parseInt(st.nextToken());
            reverse(from, to, N);
        }
        for (int i = 1; i <= N; i++) {
            System.out.print(arr[i] + " ");
        }
    }

    private static void reverse(int from, int to, int N) {
        int[] arr2 = new int[N + 1];
        for (int i = 1; i < from; i++) {
            arr2[i] = arr[i];
        }
        for (int i = from; i <= to; i++) {
            arr2[i] = arr[to - i + from];
        }
        for (int i = to + 1; i <= N; i++) {
            arr2[i] = arr[i];
        }
        arr = arr2;
    }
}

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.kt
import java.util.*

fun main() {
    var scanner = Scanner(System.`in`)
    var N = scanner.nextInt()
    var M = scanner.nextInt()
    var arr: Array<Int> = Array(N + 1) { i -> i }
    for (i in 0 until M) {
        var start = scanner.nextInt()
        var end = scanner.nextInt()
        arr = reverse(arr, start, end)
    }
    for (i in 1..N) {
        print("${arr[i]} ")
    }

}

fun reverse(arr: Array<Int>, start: Int, end: Int): Array<Int> {
    var copy = arr.clone()
    var len = end - start
    for (i in 0..len) {
        copy[start + i] = arr[end - i]
    }
    return copy
}
